controllers/User/userController.js

```diff
@@ -0,0 +1,82 @@
+const validateUser = require('../../utils/validations/index')
+const bcrypt = require('bcryptjs')
+const User = require('../../models/User/userModel')
+const response = require('../../utils/libs/response')
+const Joi = require('joi')
+
+// Register Validation schema
+const registrationSchema = Joi.object({
+    firstName: Joi.string().required(), 
+    lastName: Joi.string().required(), 
+    email: Joi.string().email().required(),
+    password: Joi.string().required().min(8),
+    phone: Joi.number().required()
+})
+
+// login Validation schema
+const loginSchema = Joi.object({
+    email: Joi.string().email().required(),
+    password: Joi.string().required().min(8)
+})
+
+// Authentication
+const registerUser = (req, res) => {
+    validateUser(registrationSchema)
+    // Continue with registration logic
+    //Please Note: While creating a new user, use assign from the model userId: uuid.v4() 
+}
+
+const loginUser = (req, res) => {
+    validateUser(loginSchema)
+    // Continue with login logic
+}
+
+
+const resetPassword = async (req, res) => {
+    // Authentication (req.user), then get mail from hash for findUser function below
+    // Put all the reset password functions under the authorization check (req.user)
+    const {userId, email, oldPassword, newPasswordOne, newPasswordTwo } = req.body
+    if(newPasswordOne !== newPasswordTwo){
+        return res.status(400).json({ message: "Password does not match" })
+    }
+    const findUser = await User.findOne({ email })
+    const isValid = await bcrypt.compare(oldPassword, findUser.password)
+    if(!isValid) return res.status(400).json({ message: "Invalid password input" })
+    const salt = await bcrypt.genSalt(10)
+    const newPasswordHash = await bcrypt.hash(newPasswordOne, salt)
+    await User.updateOne(
+        {userId: userId },
+        {$set: {password: newPasswordHash}},
+        { new: true }
+    )     
+    return response.successResMsg(res, 201, { message: "Password was successfully updated" })
+}
+
```

Comment:
Reset password flow does not work like this, but still we can use it, then we will be refractoring later.

Reply:
Yea..